Src/LiquidProjections.Testing/MemoryEventSource.cs

```diff
@@ -23,36 +25,71 @@ public MemoryEventSource(int batchSize = 10)
 
         public IDisposable Subscribe(long? lastProcessedCheckpoint, Subscriber subscriber, string subscriptionId)

```

Comment:
I'm a bit lost in those three different subscription methods. Either we need to carefully document them, or see if we can simplify the API. Maybe we can deprecate the old one, and have a single method that has a parameter to either wait or don't wait for the subscriber to catch up.